x-pack/plugin/async-search/src/test/java/org/elasticsearch/xpack/search/BlockingQueryBuilder.java

```diff
@@ -115,4 +119,51 @@ protected int doHashCode() {
     public String getWriteableName() {
         return NAME;
     }
+
+    static class QueryLatch implements Closeable {
```

Comment:
I would try to document what this class does and how to use it. I will try with my words but correct me if I am wrong: BlockingQueryBuilder blocks on each shard, until the next set of results is requested through reacquireBlock. Also failures are sent back based on the provided number of failures.

I would also consider renaming tryBlock and reacquireBlock to reflect what they are used for.

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/56026/commits/0304ec6b3b521e2ebe47c40a9c064797203bc021